### PR TITLE
fix: rename config-file to config

### PR DESCRIPTION
**Pull request by glihm**
Rework the slot CLI to always accept arguments, and wrapping them into a configuration file to only send this configuration file to the infra.

1. Currently, the infra is not returning the actual file used for deployment in the response:
```
[cli/src/command/deployments/create.rs:152:17] &config = CreateDeploymentCreateDeploymentOnKatanaConfig {
    config_file: None,
}
```
Not an issue, but we may not rely on that then.

2. The server feature from the `torii-cli/katana-cli` is not used, hence all options that are not supported by the infra (like changing the JSON RPC server etc..) are not likely to interfere with the infra settings.

3. Cleanup on graphql required with the exact path to follow to ensure correct generation. I've added the `CONTRIBUTING.md` for this purpose to keep the `README.md` informational.

4. By introducing the `server` feature for `katana-cli` and `torii-cli`, we're ensured that any functionalities that are server related are not exposed here, since the infrastructure is in charge of exposing them.

### CODE REVIEW

**Review by kariy — COMMENTED** [review]
now that we allow using all the katana flags, including the forking config, this would make `slot deployments fork` redundant, no? i dont remember why we have a specific command for forking though.

File: cli/src/command/deployments/create.rs [hunk 1]
```diff
@@ -183,6 +150,25 @@ impl CreateArgs {
...
+                }
+            }
+            KatanaConfig(_) => {
+                // Once the infra returns the config, print it.
+                if let Some(config) = service_toml {
+                    super::print_config_file(&config);
+                }
+            }
+        }
+
```
Comment (kariy) [inline cli/src/command/deployments/create.rs:171]: should we print the config on creation/update ? maybe this should just be on `slot deployments describe` ?

Comment (glihm) [reply cli/src/command/deployments/create.rs:171]: Yeah good point. I took back the idea that was already in place by outputting the config used to create the deployment. But definitely better if describe shows this only. 👍 

@steebchen sounds good to you too?

Comment (glihm) [reply cli/src/command/deployments/create.rs:171]: @steebchen for now though, seems that we don't return the config file. Will be necessary to be done to ensure slot CLI can deploy it as it's currently doing. 👍

**Comment by kariy** [comment]
> I was also surprised seeing both, and I don't know either why such separation. Maybe we can remove it from now?

iiuc, the idea is to use a centralized db (hosted by us) where the forked network (mainnet/testnet) data can be forked from. so user dont have to manually specify the forked provider url (we will use our own provider). so we should probably keep it and remove the forked options from the katana args.

File: cli/src/command/deployments/create.rs [hunk 2]
```diff
@@ -134,50 +151,21 @@ impl CreateArgs {
...
             CreateServiceCommands::Katana(_) => "katana",
             CreateServiceCommands::Torii(_) => "torii",
             CreateServiceCommands::Saya(_) => "saya",
         };
 
+        println!(
+            "\nTo view the deployment configuration, run `slot deployments describe {} {}`",
+            self.project, service
+        );
+
```
Comment (kariy) [inline cli/src/command/deployments/create.rs:168]: Should we mention this? Doesn't seem necessary to me as most users should already be somewhat aware of their deployments settings (based on the flags they provide).

Comment (glihm) [reply cli/src/command/deployments/create.rs:168]: Sounds good, will remove it.

**Comment by glihm** [comment]
> > I was also surprised seeing both, and I don't know either why such separation. Maybe we can remove it from now?
> 
> 
> 
> iiuc, the idea is to use a centralized db (hosted by us) where the forked network (mainnet/testnet) data can be forked from. so user dont have to manually specify the forked provider url (we will use our own provider). so we should probably keep it and remove the forked options from the katana args.
> 
> 
> 
> perhaps to not overbloat the pr, we can address this iteratively until we cut the next release

Makes sense in this case, I didn't know about this and that's a great idea.

Yep I'll revert to add back the fork command.

Is this fork feature actually working now?

**Comment by kariy** [comment]
> > > I was also surprised seeing both, and I don't know either why such separation. Maybe we can remove it from now?
> > 
> > 
> > 
> > iiuc, the idea is to use a centralized db (hosted by us) where the forked network (mainnet/testnet) data can be forked from. so user dont have to manually specify the forked provider url (we will use our own provider). so we should probably keep it and remove the forked options from the katana args.
> > 
> > 
> > 
> > perhaps to not overbloat the pr, we can address this iteratively until we cut the next release
> 
> Makes sense in this case, I didn't know about this and that's a great idea.
> 
> Yep I'll revert to add back the fork command.
> 
> Is this fork feature actually working now?

Unsure

File: cli/src/command/deployments/create.rs (hunk 2 above)
Comment (glihm) [reply cli/src/command/deployments/create.rs]: 06e3794363af1d76236f9af394d3822b87a357d3

**Comment by glihm** [comment]
> > I was also surprised seeing both, and I don't know either why such separation. Maybe we can remove it from now?
> 
> iiuc, the idea is to use a centralized db (hosted by us) where the forked network (mainnet/testnet) data can be forked from. so user dont have to manually specify the forked provider url (we will use our own provider). so we should probably keep it and remove the forked options from the katana args.
> 
> perhaps to not overbloat the pr, we can address this iteratively until we cut the next release

Reverted in 8a1d19ae5302a87ccf6d257434453299299acd02

**Pull request merged**